### PR TITLE
Poster edit: Add reviews tab

### PR DESCRIPTION
**Pull request by slax57**
## Content

- [x] Add 'Reviews' tab with ReferenceManyField and DataGrid
- [x] Customize DataGrid options and style
- [x] Add pagination
- [x] Rebase onto https://github.com/slax57/admin-e-commerce-demo/pull/9

## Demo
2022-03-24 15_03_41-Vite App

### CODE REVIEW

File: src/product/ProductEdit.tsx [hunk 1]
```diff
@@ -106,6 +123,54 @@ const ProductEditForm = () => {
...
+                sx={{
+                  overflow: "hidden",
+                  maxWidth: "20em",
+                  whiteSpace: "nowrap",
+                  textOverflow: "ellipsis",
+                }}
+              >
+                <TextField source="comment" />
+              </Box>
+            </WrapperField>
```
Comment (djhi) [inline src/product/ProductEdit.tsx]: Why not just a `TextField` with the `sx` prop?

File: src/product/ProductEdit.tsx [hunk 2]
```diff
@@ -106,6 +123,54 @@ const ProductEditForm = () => {
...
+              width: "100%",
+            }}
+          >
+            <DateField source="date" />
+            <ReferenceField source="customer_id" reference="customers">
+              <CustomerField />
+            </ReferenceField>
+            <WrapperField label="Rating" sortBy="rating">
+              <Rating />
+            </WrapperField>
```
Comment (djhi) [inline src/product/ProductEdit.tsx]: I didn't even know about this `WrapperField`. I think you don't need it though. Just accept both props on your component event though you don't use them internally

**Review by djhi — CHANGES_REQUESTED** [review]

File: src/product/ProductEdit.tsx (hunk 1 above)
Comment (slax57) [reply src/product/ProductEdit.tsx]: Didn't work when I tried it. Text overflow props need to be declared in the parent node to work 🤷

Comment (slax57) [reply src/product/ProductEdit.tsx]: As you suggested, I created a `EllipsisTextField` component to wrap this 🙂

**Review by djhi — APPROVED** [review]

**Pull request merged**